prov/efa/src/efa_domain.c

```diff
@@ -133,7 +133,11 @@ static int efa_domain_init_rdm(struct efa_domain *efa_domain, struct fi_info *in
 	efa_shm_info_create(info, &efa_domain->shm_info);
 
 	if (efa_domain->shm_info) {
-		assert(!strcmp(efa_domain->shm_info->fabric_attr->name, "shm"));
+		if (rxr_env.use_sm2) {
```

Comment:
can we use a char * shm_provider_name to refer to either shm or sm2 and use a single assert here.